libs/core/langchain_core/tools.py

```diff
@@ -439,23 +421,45 @@ def raise_deprecation(cls, values: Dict) -> Dict:
             values["callbacks"] = values.pop("callback_manager", None)
         return values
 
-    @abstractmethod
-    def _run(
-        self,
-        *args: Any,
-        **kwargs: Any,
-    ) -> Any:
+    def _run_include_raw_output(self, *args: Any, **kwargs: Any) -> Tuple[Content, Any]:
+        """Return a repr of the result to pass to a model and the raw_output result."""
+        if not signature(self._run).parameters.get("run_manager"):
+            kwargs.pop("run_manager")
+        return self._run(*args, **kwargs), None
+
+    async def _arun_include_raw_output(
+        self, *args: Any, **kwargs: Any
+    ) -> Tuple[Content, Any]:
+        """Return a repr of the result to pass to a model and the raw_output result."""
+        if (
+            self.__class__._run_include_raw_output
+            is not BaseTool._run_include_raw_output
+        ):
+            if "run_manager" in kwargs:
+                kwargs["run_manager"] = kwargs["run_manager"].get_sync()
+            return await run_in_executor(
+                None,
+                self._run_include_raw_output,
+                *args,
+                **kwargs,
+            )
+        else:
+            if not signature(self._arun).parameters.get("run_manager"):
+                kwargs.pop("run_manager")
+            return (await self._arun(*args, **kwargs)), None
+
+    def _run(self, *args: Any, **kwargs: Any) -> Any:
```

Comment:
Given that `_run` is basically untyped, what are your thoughts on adding an attribute instead to the tool that specifies the expected return format of `_run`. That way we don't need to introduce a new method

Reply:
think this is mostly a question of minimizing contributor friction, down for whatever makes contributor life easiest

Reply:
I think a single `_run` with an attribute would be easier to understand